javaparser-core/src/main/javacc/java_1_8.jj

```diff
@@ -153,6 +153,16 @@ final class ASTParser {
 	    return ret;
     }
 
+    private ReferenceType juggleArrayTypes(Range range, Type type, int arrayCount, List<List<AnnotationExpr>> arrayAnnotations) {
+        if (arrayCount == 0) {
+            return new ReferenceType(range, type);
+        }
+        for (List<AnnotationExpr> annotations: arrayAnnotations) {
+            type = new ArrayType(range, type, annotations);
```

Comment:
Annotations are parsed from left to right, thus instantiations of ArrayTypes should happen from right to left. (c.f. example in JLS section 9.7.4)